src/pudl/extract/excel.py

```diff
@@ -267,7 +267,10 @@ def extract(self, **partitions):
                 # check if there are any missing or extra columns
                 str_part = str(list(partition.values())[0])
                 col_map = self.METADATA._column_map[page]
-                page_cols = col_map.loc[col_map[str_part].notnull(), [str_part]].index
+                page_cols = col_map.loc[
+                    (col_map[str_part].notnull()) & (col_map[str_part] != -1),
+                    [str_part],
+                ].index
```

Comment:
Basically what was happening here was that any time you extracted the data you would get a bunch of warnings that columns were missing when we didn't expect them to be here. By adding a filter for the rows with -1 we no longer get those warnings.

Reply:
hey i think this is giving the warnings with a bunch of columns that are unmapped here for good reason. Basically the check is finding all of the columns that *do* exist in the data being extracted but *do not* exist in the column maps (which is to say have -1's as values).

I believe I edited this check during the last round of updates to the EIA data because it was previously hard to tell whether everything was actually mapped during #1836. Because we were focused on the eia datasets in the etl, we totally missed doing the complete map for 861. we should probably make an issue for it to get in there and actually map them.

Reply:
I looked at a couple of examples, and I think that it's doing the opposite. Without the `& (col_map[str_part] != -1)` that I added the warning shows values that _don't_ exist in the raw data but _do_ exist in the expected cols. As far as I can tell, the columns marked with -1 in the column map are not in the raw data and therefore shouldn't be part of the expected columns. We _do_ want to see instances where columns in the raw data aren't covered in the expected columns, but the only way to do that is to make the expected columns indicative of what we actually expect to find in a given year of data. We also want to know if there are cases where an expected column doesn't appear in the raw data, but I'm fairly certain that I added the -1 values to the expected columns because the column wasn't there.

Here's a code snipit for context

```
str_part = str(list(partition.values())[0])
col_map = self.METADATA._column_map[page]
page_cols = col_map.loc[
    (col_map[str_part].notnull()),
    [str_part],
].index
expected_cols = page_cols.union(self.cols_added)
if set(newdata.columns) != set(expected_cols):
    extra_raw_cols = set(newdata.columns).difference(expected_cols)
    missing_raw_cols = set(expected_cols).difference(newdata.columns)
    if extra_raw_cols:
        logger.warning(
            f"Extra columns found in extracted table of "
            f"{page}/{str_part}: {extra_raw_cols}"
        )
    if missing_raw_cols:
        logger.warning(
            "Expected columns not found in extracted table of "
            f"{page}/{str_part}: {missing_raw_cols}"
```

Reply:
ah no i think you are correct here. maybe the `notnull()` was previously trying to do the same `!= -1` thing but it definitely is not and it doesn't look like we ever convert the -1's to nulls so i think you are correct here.